src/common/hooks/useConfig/useConfig.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { renderHook, act } from "@testing-library/react-hooks";
+import { useConfig } from "./index";
+import { AsyncStorage } from "react-native";
+
+jest.mock("react-native", () => ({
+  AsyncStorage: {
+    setItem: jest.fn(),
+    getItem: jest.fn()
+  }
+}));
+
+const mockGetItem = AsyncStorage.getItem as jest.Mock;
+const mockSetItem = AsyncStorage.setItem as jest.Mock;
+
+describe("useConfig", () => {
+  beforeEach(() => {
+    mockGetItem.mockReset();
+    mockSetItem.mockReset();
+    mockGetItem.mockResolvedValue(JSON.stringify({ network: "mainnet" }));
+  });
+  it("should load saved config automatically", async () => {
+    expect.assertions(2);
+    const { result, waitForNextUpdate } = renderHook(() => useConfig());
+
+    await act(async () => {
+      await waitForNextUpdate();
+      await expect(result.current.loaded).toBe(true);
+      await expect(result.current.config).toStrictEqual({ network: "mainnet" });
+    });
```

Comment:
I think this should work:
```
    await waitForNextUpdate();

    expect(result.current.loaded).toBe(true);
    expect(result.current.config).toStrictEqual({ network: "mainnet" });
```

Reply:
Need the act cause the state changes because of the useEffect hook.